runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/UnboundedDataset.java

```diff
@@ -57,15 +60,20 @@ public UnboundedDataset(JavaDStream<WindowedValue<T>> dStream, List<Integer> str
     return streamSources;
   }
 
-  public void cache() {
-    dStream.cache();
+  @SuppressWarnings("unchecked")
```

Comment:
Perhaps we can take this opportunity to consolidate the two `cache` methods here to one, no real need for this extra delegation to another method here.

Reply:
right